tests/gis_tests/gdal_tests/test_raster.py

```diff
@@ -581,8 +582,9 @@ def test_band_data(self):
             self.assertAlmostEqual(self.band.mean, 2.828326634228898)
             self.assertAlmostEqual(self.band.std, 2.4260526986669095)
 
-            # Statistics are persisted into PAM file on band close
+            # Statistics are persisted into PAM file on band close when the object is garbage collected.
             self.band = None
+            gc.collect()
```

Comment:
If possible I think it would be preferable to avoid calling `gc.collect()`. There is a chance that GDAL writes the metadata file only when the raster is closed, and maybe removing only the band does not imediately close the GDALRaster? Could we try to add a reference to the raster in the setUp like `self.raster = GDALRaster(self.rs_path)` and then call `self.raster = None` instead of `self.band = None` to see if that makes a difference?

There is also a [GDAL_PAM_ENABLED](https://trac.osgeo.org/gdal/wiki/ConfigOptions#GDAL_PAM_ENABLED) flag that might be set to False in the system with failing tests. So for this test we could also add something like `skipIf(os.env.get('GDAL_PAM_ENABLED'))`.